prover/prover_fri/src/utils.rs

```diff
@@ -118,12 +119,18 @@ pub async fn save_proof(
             .await;
     }
     if job_metadata.is_node_final_proof {
+        let circuit_id = if job_metadata.circuit_id == EIP_4844_CIRCUIT_ID {
+            EIP_4844_CIRCUIT_ID
+        } else {
+            get_base_layer_circuit_id_for_recursive_layer(job_metadata.circuit_id)
+        };
```

Comment:
I don't love this but given that it'll be gone with 1.5.0 I think it's the right compromise